src/h5web/providers/hsds/utils.ts

```diff
@@ -71,6 +71,14 @@ export function convertHsdsType(hsdsType: HsdsType): HDF5Type {
 
   switch (hsdsType.class) {
     case HDF5TypeClass.Enum:
+      // Booleans are stored as Enum by h5py
+      // https://docs.h5py.org/en/stable/faq.html#what-datatypes-are-supported
+      if (hsdsType.mapping === { FALSE: 0, TRUE: 1 }) {
+        return {
+          class: HDF5TypeClass.Bool,
+        };
+      }
```

Comment:
As I said, boolean types are now a special type in h5web. As HSDS returns them as an enum with a specific mapping, I check the mapping here to return the appropriate h5web type.

Reply:
This can't possibly work, since you're comparing two different object references. Maybe just check for `hsdsType.mapping.FALSE === 0` -- seems sufficient.

Reply:
I indeed realized this as I was writing tests :smile: